File: app/src/main/java/dev/ragnarok/fenrir/mvp/view/IAudioCatalogView.java
package dev.ragnarok.fenrir.mvp.view;

import java.util.List;

import dev.ragnarok.fenrir.model.AudioCatalog;
import dev.ragnarok.fenrir.mvp.core.IMvpView;
import dev.ragnarok.fenrir.mvp.view.base.IAccountDependencyView;


public interface IAudioCatalogView extends IAccountDependencyView, IMvpView, IErrorView {
    void displayData(List<AudioCatalog> pages);

    void notifyDataSetChanged();

    void notifyDataAdded(int position, int count);

    void showRefreshing(boolean refreshing);
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/mvp/view/IAudioCatalogView.kt
package dev.ragnarok.fenrir.mvp.view

import dev.ragnarok.fenrir.model.AudioCatalog
import dev.ragnarok.fenrir.mvp.core.IMvpView
import dev.ragnarok.fenrir.mvp.view.base.IAccountDependencyView

interface IAudioCatalogView : IAccountDependencyView, IMvpView, IErrorView {
    fun displayData(pages: List<AudioCatalog>)
    fun notifyDataSetChanged()
    fun notifyDataAdded(position: Int, count: Int)
    fun showRefreshing(refreshing: Boolean)
}